src/database/dbutils.go

```diff
@@ -0,0 +1,111 @@
+package database
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"github.com/jackc/pgx/v5"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close(context.Context) error
+}
+
+func establishConnection(ctx context.Context) (*pgx.Conn, error) {
+	conn, err := pgx.Connect(ctx, os.Getenv("DATABASE_URL"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to database: %w", err)
+	}
+	return conn, nil
+}
+
+func insertRows(ctx context.Context, conn PgxIface, index *pb.Index) error {
+
+	tx, err := conn.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("unable to connect to database: %w", err)
+	}
+
+	_, err = tx.Exec(ctx, `
+		CREATE TABLE IF NOT EXISTS index_list (
+			name TEXT PRIMARY KEY,
+			uri TEXT,
+			data_type TEXT,
+			source_type TEXT,
+			first_added_time TIMESTAMP,
+			last_refreshed_time TIMESTAMP,
+			content TEXT,
+			word_occurrences JSONB
```

Comment:
Can you investigate whether there are other/better datatypes than JSONB for a string->int map like word_occurrences? Or is JSONB good enough because it allows for direct indexing/subqueries without eg deserializing the data?

Reply:
I found that the best way to do this is to use JSONB as we can index it directly in PostgreSQL